### ISSUE TITLE
PMS for Mac not picking up or syncing new network path

### ISSUE THREAD

**Issue by ptvemail** [croneter/PlexKodiConnect#298]
Hello.  I'd be very grateful if I could get some help with setting up PKC.  I believe in its benefits and I appreciate the work that has gone into it, and all the support I've seen given freely on the forums.  

I'm trying to set up direct paths from a Mac SMB server, and I've followed the set-up guides (including adding those paths to the PMS and disabling UNC and plex paths in the PKC settings) but to no avail.  I keep getting a PKC sync error (telling me "Kodi can't locate file: ......and it shows the ACTUAL path of the file on my Mac, and not the smb path, even when the SMB path is the ONLY path now within PMS).

My work so far:
I've worked hard on this (4 days), read the FAQ and known issues, read forums for hints, and have tested too many combinations of things.  I started off trying to get this to work on Kodi 16.1, then switched to SPMC hoping that'd help, but I still get the same sync path issues.  I've been resetting SPMC prior to every test to ensure I try things cleanly.  I'm hopeful I've just done something (systematically) silly, and I'm open to being corrected.

Set-Up:
I have an iMac running Sierra, and I set up an SMB server on it using the instructions on your site.  I set up a static IP for it (on the mac, not on the router).  The guest user on the Mac is enabled and guests can connect to shared folders.  I want to use direct paths, so I followed your "substitute paths on PMS" instructions and added the SMB path to my Plex server and on the Minix I set up PKC with both Plex UNC paths and Replace paths disabled as recommended in the instructions.  I did the SMB access test as suggested on my Minix, and yes I can reach the SMB files if I try to "add network connection" but never ever if I try it through the SMB add bit.  When I add the SMB network it fails if I put in my Mac username/pass - it seems they must be blank.  Zeroconf works also if that's relevant.  On the PKC setup, for the network credentials I've used my Mac's IP (with and without the :445, with and without my Mac's username/pass, and I've checked the file and folder permissions on the Mac).

Partial Success:
Frustratingly I had half-an-hour of relative success yesterday where the libraries synced and videos loaded pretty quickly (3+ seconds).  I saw they had complicated paths when I did an "info" request "plexkodiconnect blah blah", not smb://etc, so I am not sure if even that set-up was correct, but in any case I got cocky and thought I'd solved all my troubles, did a reset so I'd have a clean installation and now cannot replicate my relative success.  Perhaps it's relevant, but I used my Mac's NetBIOS name and user account for the network credentials in that "successful" case, but the NetBIOS on my Mac has since changed to Null for some reason, so I've been using my Mac's IP for the network credentials since then (plus user name/pass).  I've also tried the "Computer Name" and "Computer Name".local (with user/pass and without) but no luck there either.

Samba Access tests:
The Mac's SMB share (and PMS and "null") can be seen and used well by VLC on my Android phone and VLC on my tablet, and both VLCs show the static IP I set up in the folder heading (with :445). As stated above SPMC on the Minix can see the SMB folder easily if I DON'T add a user/pass in the "add network connection" video file browser, however interestingly VLC on the Minix does NOT find the Mac's SMB server.  I also think the SMB connections are dying when I reboot the Minix ("broken pipe").

Other details:
Minix is up to date (but still on an old O/S - updated in Jan).  Try to wake up remote servers is ticked in SPMC, and I've experimented with putting the Minix's IP in the PMS "List of IP addresses and networks that are allowed without auth" thing but to no effect. 

Conclusion/request:
I'd be most appreciative of any help.  Clearly I'm too deeply involved in it and I should have paused to reflect and seek help.  I apologise if the above is too much info, and am sure there'll be a need to clarify, which is fine.  I'm attaching both a current and the previous "old" log joined together in case they're helpful.  I've removed the token information as suggested and have replaced a few other details for good measure (plex account name and number and Mac IP).  Kind regards, Peter.

**Comment by croneter** [comment]
Let's make an exception and help you here (I wouldn't want people opening Github issues to individually help them in setting up Direct Paths 🤔 )

Seems to me like this is a (smb) file permissions issue. Kodi is a bit peculiar there...

1. Do install Kodi 17.1, I don't see a reason why you wouldn't want to do that.
2. Start from scratch with Kodi. 
    * This is really important if you entered wrong network credentials in the past and you don't manage to erase them. 
    * To do that, simply delete your entire `userdata` folder when Kodi is not running: http://kodi.wiki/view/Userdata
3. Follow http://kodi.wiki/view/Adding_video_sources
    * In Step 5 _Browse for new share_, select `Network File System`. Try browsing to your media.
    * You MIGHT get prompted for user/password, but probably not since you enabled to guest account
    * If you manage to browse to e.g. a movie's location, hit `OK`
    * In Step 6 _Add Video Source_ hit `Cancel`
4. Install PKC. At the end of the initial first-startup-dialogs, choose to enter the PKC settings
5. In PKC, set the PKC settings according to https://github.com/croneter/PlexKodiConnect/wiki/Option-2-Substitute-Paths-on-the-PMS 
6. Restart Kodi

If this does not help, please be specific in what you can't achieve. Also, using the Plex web interface, navigate to a movie, hit the three dots ..., then Get Info, then Download XML (or something akin). Please share this XML here

**Comment by ptvemail** [comment]
Hello.  Thank you for your kind and thoughtful reply, and for just replying in the circumstances.  I hadn’t acquainted myself with the Github etiquette, and I should have.

Thanks also for the insights about where I might be going wrong.  Your mentioning that it might be a SMB permissions issue, and your suggestion of adding an NFS mount got me checking my Mac….more on that in a minute…

First though, I cannot use 17.1 yet, as my Minix is still on an older Android.  I believe the Minix people are going to put out an update to enable people to use 17.1, but it’s not out yet.  

I reset Kodi (SPMC actually) using Android’s App manager (deleting App storage) and that did the same thing (I checked the directories to be sure).

I then tried to add an NFS mount.  SPMC just thought about it (circle on screen) then bombed out.  …but it got me thinking about my network again…so I returned to my Mac.  I noticed that under “file sharing” on the mac it said it was sharing files at “smb://(null).gateway”…I had noticed, and then overlooked that yesterday…it seemed to get that way sometime during that day.  So this evening I realised if I switched the ethernet connection on the mac from DHCP with a manual address to just DHCP, and then back to DHCP with a manual address again it’d correct the smb path in the mac “sharing” setting page.  

Since then I’ve (reset SPMC) then tried NFS again, but still no luck.  I’d like to use NFS, but it’s not a huge deal.

Anyway, since my DHCP trick VLC on my Minix now has a “Mac” samba server, not “null”.  That’s a promising sign.  Mac’s sharing has the static IP address back in the header, and my NetBios name is also back to “Mac” (it was also null when I first wrote).  So perhaps SMB might work now.  It’s midnight here and I better rest.  I’ll have a go in around 24 hrs from now and post back with the XML and more specific details as you requested if necessary.

I really appreciate your message, and my apologies again for going against the usual posting practice. 

kind regards
Peter.

**Comment by croneter** [comment]
> I cannot use 17.1 yet, as my Minix is still on an older Android
If you're unlucky, your issues stem from this. SMB implementations have been wonky in the past

> I reset Kodi (SPMC actually) using Android’s App manager (deleting App storage)
That's OK 😉 

* Could you use another Mac or PC with Kodi installed and try to access your Mac shares (NOT using your Minix)
    * Until you cannot successfully do that, your Mac SMB shares are not set-up correctly
* How does your SMB share look like now? So e.g. `smb://myServer/Plex`

**Comment by ptvemail** [comment]
Hi again, 

it’s super awesome of you to provide your thoughts, I really appreciate it.  

I just had a go again with a SMB connection.  No luck.  I had hoped my fixing the Netbios and filesharing addresses on my Mac SMB sharing would help, but oh well.  As mentioned last night the SMB share now appears nicely on the Minix’s VLC, as well as on my phone and tablet (I can go on to play the videos too in VLC on the Minix and phone).  So near but yet so far.  Still, they’re just first world problems, it’s ok.

* If you're unlucky, your issues stem from this. SMB implementations have been wonky in the past
Oh. In that case I’ll just cease my efforts on direct paths and reconfigure PKC when I get an Android update that allows me to update Kodi.  Thanks for that helpful info.

* Could you use another Mac or PC with Kodi installed and try to access your Mac shares (NOT using your Minix)
No mate, I don’t have one.  Oh actually, I could try my girlfriend’s Mac maybe.  Anyway, the other thing I could do is try SPMB on my galaxy s7 phone (android 7, so allows me to use the latest Kodi) and/or my older tablet (old Kodi).   That could test the wonky old Kodi SMB and the new Kodi SMB - does that seem like a prudent test?  Thanks for the idea to try other machines!  

		• Until you cannot successfully do that, your Mac SMB shares are not set-up correctly
	• How does your SMB share look like now? So e.g. smb://myServer/Plex <smb://myServer/Plex>
- OK, I can add the SMB server on SPMC/Kodi (via video, files, browser etc) provided I use no password (I get broken pipes or connection refused if I use the password), and the address for that is as you describe above - smb://blah.blah.blah.blah <smb://blah.blah.blah.blah>
- When I access the SMB server via my android phone’s VLC, it reports the same IP address but with 445 at the end - so [blah.blah.blah.blah]:445
- in my PMS I have smb://blah.blah.blah.blah:445/Travel <smb://blah.blah.blah.blah:445/Travel> as a path for my “Travel” shared folder. 
(WAIT!!!  - I did have an additional / before smb …I just realised that in my proof-read of this before sending…I’ll try again tomorrow, it’s late)

- in Mac sharing it says: Computers on your local network can access your computer at:MAC.local and Other users can access shared folders on this computer, and administrators all volumes, at “afp://mac.gateway” or “smb://mac.gateway”.     

….hmmm….just wondering if I should have put mac instead of MAC (MAC is in the Netbios and in the above address) in the network credentials step

…also, your notes for the option 2 installation don’t talk about entering the network credentials…I assume that was an oversight, but I tried not entering them too, but that didn’t help.  

Also, in your instructions, you say:
In Kodi, start adding a Video source following this guide <http://kodi.wiki/view/Adding_video_sources> from step 1 to 6 only - NOT to the end!
If you manage to navigate to your media, it is indeed accessible to Kodi
Hit Cancel in step 6
can I confirm with you that you mean to hit ok at the end of step 6 (after you have named the folder “Movies” or whatever, and then cancel before step 7 "This directory contains"
  
sorry, I hadn’t meant to write another long one.  As you see I have a few things to test yet.  Again, I really appreciate the leads.  I’ll let you know how I go tomorrow evening, thanks!

**Comment by croneter** [comment]
Dude, this gets a bit convoluted 🙈 . Try to do one thing at a time. Try to be brief and concise in your replies 😆 . Until you can't solve a step, do not go to the next one
1. Using the Plex web interface, navigate to a movie, hit the three dots ..., then Get Info, then Download XML (or something akin). Please share this complete XML here
2. Make sure that your media is somehow accessible in the network from another computer (be that your Kodi machine, your gf's mac or whatever)
3. Follow steps 1...6 in my reply above

**Comment by croneter** [comment]
> can I confirm with you that you mean to hit ok at the end of step 6 (after you have named the folder “Movies” or whatever, and then cancel before step 7 "This directory contains"

Yes that's fine

**Comment by ptvemail** [comment]
OK, that makes sense, my apologies.  I’ll do as you say later today and will be in touch then, thanks again

**Comment by ptvemail** [comment]
Hi, I haven't had the chance to test things again today, but here's an XML file.  

I realise I need to slow down on this, and I'm just generally tired from a busy week.  I have Friday off work.  I'll try again then. 

Thanks so much.  

[259114.pdf](https://github.com/croneter/PlexKodiConnect/files/1007756/259114.pdf)

**Comment by croneter** [comment]
Thanks for the XML. There is always a file entry in these XMLs; in your case:
```
file="/Volumes/LaCie/Minix-TV/Travel/The World From Above - Sicily.mp4"
```
If you follow https://github.com/croneter/PlexKodiConnect/wiki/Option-2-Substitute-Paths-on-the-PMS correctly, you will **NOT** see a local path like `/Volumes/...` but something like `smb://myServer/Travel/The World From Above - Sicily.mp4`

Do relax now 😝

**Comment by ptvemail** [comment]
Haha thanks, I really need to chill, you know that.  

Thanks for that!!!  I was chasing the wrong problem then evidently.  I had thought I had followed the instructions properly, but clearly not.  I’ll chill out and have another go tomorrow - thanks so much!

**Comment by ptvemail** [comment]
OK, so I’ll just focus on one aspect for now - getting the right path into my PlexMS XML files.  Thanks very much for the help so far.  

I’ll succinctly (:)) show my work at every step below.  I suspect my typing the smb address in (middle of step 5 below) isn’t what you want…that I must select it somehow…

1. Setup directories shared over SMB on the server
done 

2. Check that your media is indeed accessible to Kodi
reset kodi, added smb server using “Browse” feature, could navigate to a folder within it, cancelled before step6

3. Make tripple sure ...
other tests: files accessible via VLC on phone, tablet, minix and accessible via kodi smb path on minix

4. For each of your Plex libraries, note the new network path. 
Our example Plex library here:  Linux/Mac Server:  smb://myNas/movies
Done - mine shows the IP - smb://IP.AD.DRE.SS/Documentaries/ for example (the slash at the end appears in Kodi)

5. start following the Moving Content to a new Location guide by Plex
disable the Empty trash automatically…. done
Quit or exit the Plex Media Server….  
Copy your media files to the desired location  - not done, as per your instructions
Once your content is available on the new location, start your Plex Media Server (I assume there’s no need to start and then stop if there’s no copy in the middle)
Open your Plex Web App …done
Edit a Library and then add the appropriate folder for where your content is now located (I go to the “Add Folder” section - my smb server does NOT show up on the left hand side as a disc to select, so I’ve typed smb://IP.AD.DRE.SS/Documentaries <smb://IP.AD.DRE.SS/Documentaries> here (leaving off the end slash, because the current path doesn’t have one))
Update the Library - done
(I still have two paths in my PMS for each source at this point, and the XML still has the physical path of the disc drive in the filename at this point, not smb://….)
 remove the old/previous folder location - done, plex updates library automatically, red trash cans appear above my library posters, I do a manual update to be sure, XML still has drive path not smb
you may need to empty your Library trash in order to remove the entries where the content was stored at the old content location - emptied, all items in library vanish (“there are no items in this library”)
I’d love your thoughts, but respect your time. Just if possible.  I’ve already decided to donate regardless of whether we get this going or not, I appreciate your help.

**Comment by croneter** [comment]
OK, your step 5. is where things go sour, apparently. Very weird that the PMS won't pick up the new smb paths. I'm sorry, but I think I can't help you further, since this is a PMS thing.

Could you please take this issue to the Plex Forums and the PMS for Mac section? If you did not screw things up (I don't think you did), this is a PMS bug 😒

**Comment by croneter** [comment]
## Workaround
* Use PKC Addon Paths instead of Direct Paths
* Use PKC to substitute the PMS paths ([see here](https://github.com/croneter/PlexKodiConnect/wiki/Option-1-Substitute-Paths-using-PKC)). This is not always possible, depending on your library and PMS ssetup

**Comment by croneter** [comment]
I think it would help if you shared a couple of Screenshots for your last step 5, if you find the time

**Comment by ptvemail** [comment]
OK thanks, in a way I’m glad it’s a bug…

…anyway, I’ll have a go at the option 1 setup, and take some screenshots of step5 before that and send them.  I’ll do that tomorrow, it’s late here

all the best and thanks again

**Comment by ptvemail** [comment]
Hi again.  Here is my Plex forum post:
[https://forums.plex.tv/discussion/272279/pms-for-mac-not-picking-up-or-syncing-new-network-path](url)

I'll refrain from trying to set up Option 1 until I've had some responses to my Plex forum post.  

Aside: I'd respectfully recommend an "Other Videos" capability for Option 1 too if it's easy to do.  I have quite a lot of TV shows with manually added posters/metadata, and if I merge them in with my TV shows files (which have auto metadata) I'm concerned I'll lose my manual metadata.

thanks again for your tremendous help.  I'll let you know if someone comes up with something.

**Comment by croneter** [comment]
You need to mount network shares when you use a PMS on either Mac or Linux. See the updated wiki for a how-to: https://github.com/croneter/PlexKodiConnect/wiki/Option-2-Substitute-Paths-on-the-PMS

Also, I won't add an option for "Other Videos" as PKC cannot identify them anyway. They come across as "Movies"

**Comment by ptvemail** [comment]
Hello,

thanks again for your help recently.  I apologise for lacking focus initially, and for taking so long to post this reply.  

I am pleased to report that I have direct paths working, however it is using Emby.  

To quickly recap, I was trying to run PMS on my 2014, Sierra iMac, and it couldn’t see the smb serves I had set up on the same mac.  

We discussed the need to mount the smb server, and I see you have updated your Option 2 instructions to highlight the need to do so.  

I did try to mount my smb server on my mac using finder, but that is where I got stuck.  OS X would give me the following error: “This file server is available on your computer.  Access the volumes and files locally.”  …well OS X’s accusation was true, they were on the same computer, but I could never get PMS to pick them up off smb.  I researched how to get around the Mac error, or if it was an error, and I didn’t like the hacky nature of the suggestions so I didn’t proceed.  I had no such trouble with Emby.  I just pasted the smb path of the files in the appropriate folder selection spot and it worked, so at least with my untrained eye I guess you were right when you said there might be a PMS bug. I reported it via a Plex forum, but didn’t hear back.  

Anyway, I’m going fine with Emby now, but I still appreciate your help, and so I have donated two coffees.

In case it’s helpful, it may be useful to consider adding instructions on how to use finder to mount a share.  The Option 2 instructions link to a guide from Plex focuses on making the mount permanent, and doesn’t go into much detail about how to mount the share in the first place (the Plex instructions just say to do it with finder, but not how). 

thanks again, Peter.
